src/Core/WhenExpr/WhenExpr.re

```diff
@@ -0,0 +1,156 @@
+module Value = {
+  [@deriving show({with_path: false})]
+  type t =
+    | String(string)
+    | True
+    | False;
+
+  // Emulate JavaScript semantics
+  let asBool =
+    fun
+    | True => true
+    | False => false
+    | String("") => false
+    | String(_) => true;
+
+  // Emulate JavaScript semantics
+  let asString =
+    fun
+    | True => "true"
+    | False => "false"
+    | String(str) => str;
+};
+
+[@deriving show({with_path: false})]
+type t =
+  | Defined(string)
+  | Eq(string, Value.t)
+  | Neq(string, Value.t)
+  | Regex(string, option(Re.re))
+  | And(list(t))
+  | Or(list(t))
+  | Not(t)
+  | Value(Value.t);
+
+let evaluate = (expr, getValue) => {
+  let rec eval =
+    fun
+    | Defined(name) => getValue(name) |> Value.asBool
+    | Eq(key, value) => getValue(key) == value
+    | Neq(key, value) => getValue(key) != value
+    | Regex(_, None) => false
+    | Regex(key, Some(re)) => Re.execp(re, key |> getValue |> Value.asString)
+    | And(exprs) => List.for_all(eval, exprs)
+    | Or(exprs) => List.exists(eval, exprs)
+    | Not(expr) => !eval(expr)
+    | Value(value) => Value.asBool(value);
+
+  eval(expr);
+};
+
+module Parse = {
+  // Translated relatively faithfully from
```

Comment:
I think this was the right call to switch from menhir to manual parsing for this particular case. As you mentioned, it'll be easier to keep in sync later too.